File: backend/src/dto/runner.rs
use gitdot_api::resource::runner as api;
use gitdot_core::dto::{CreateRunnerTokenResponse, RunnerResponse};

use super::IntoApi;

impl IntoApi for RunnerResponse {
    type ApiType = api::RunnerResource;
    fn into_api(self) -> Self::ApiType {
        api::RunnerResource {
            id: self.id,
            name: self.name,
            owner_id: self.owner_id,
            owner_name: self.owner_name,
            owner_type: self.owner_type.into(),
            last_verified: self.last_verified,
            created_at: self.created_at,
        }
    }
}

impl IntoApi for CreateRunnerTokenResponse {
    type ApiType = api::RunnerTokenResource;
    fn into_api(self) -> Self::ApiType {
        api::RunnerTokenResource { token: self.token }
    }
}

File: frontend/app/lib/dto/runner.ts
import { z } from "zod";

export const RunnerResponseSchema = z.object({
  id: z.uuid(),
  name: z.string(),
  owner_id: z.uuid(),
  owner_type: z.string(),
  last_verified: z.string().optional().nullable(),
  created_at: z.string(),
});

export type RunnerResponse = z.infer<typeof RunnerResponseSchema>;

export const RunnerTokenResponseSchema = z.object({
  token: z.string(),
});

export type RunnerTokenResponse = z.infer<typeof RunnerTokenResponseSchema>;
